nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -140,10 +140,11 @@ Costmap2DROS::on_configure(const rclcpp_lifecycle::State & /*state*/)
   }
 
   // Create the transform-related objects
-  tf_buffer_ = std::make_shared<tf2_ros::Buffer>(rclcpp_node_->get_clock());
+  tf_buffer_ = std::make_shared<tf2_ros::Buffer>(get_clock());
   auto timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
-    rclcpp_node_->get_node_base_interface(),
-    rclcpp_node_->get_node_timers_interface());
+    get_node_base_interface(),
+    get_node_timers_interface(),
+    callback_group_);
   tf_buffer_->setCreateTimerInterface(timer_interface);
   tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_);
```

Comment:
Didn't we add https://github.com/ros2/geometry2/blob/ros2/tf2_ros/src/transform_listener.cpp#L43 specifically so we could set `spin_thread` to `true` so that it handled its own thread for getting data? Else right now the TF buffer would be blocked by service or subscriber callbacks